connector/src/main/scala/com/microsoft/kusto/spark/utils/KustoDataSourceUtils.scala

```diff
@@ -343,78 +343,76 @@ object KustoDataSourceUtils {
     if (!storageSecretIsAccountKey) {
       // If SAS option defined - take sas
-      KustoDataSourceUtils.parseSas(storageSecret.get)
+      Some(KustoDataSourceUtils.parseSas(storageSecret.get))
     } else {
       if (storageAccount.isEmpty || storageContainer.isEmpty || storageSecret.isEmpty) {
         val keyVaultParameters = KeyVaultUtils.getStorageParamsFromKeyVault(keyVaultAuthentication)
         // If KeyVault contains sas take it
         if(!keyVaultParameters.secretIsAccountKey) {
-          keyVaultParameters
+          Some(keyVaultParameters)
         } else {
-          // Try combine
+          if ((storageAccount.isEmpty && keyVaultParameters.account.isEmpty) ||
+              (storageContainer.isEmpty && keyVaultParameters.container.isEmpty) ||
+              (storageSecret.isEmpty && keyVaultParameters.secret.isEmpty)) {
 
-          val account = if (storageAccount.isEmpty) {
-            Some(keyVaultParameters.account)
-          } else {
-            storageAccount
-          }
-          val secret = if (storageSecret.isEmpty) {
-            Some(keyVaultParameters.secret)
-          } else {
-            storageSecret
-          }
-          val container = if (storageContainer.isEmpty) {
-            Some(keyVaultParameters.container)
-          } else {
-            storageContainer
-          }
+              // We don't have enough information to access blob storage
+              None
+            }
+          else {
+            // Try combine
+            val account = if (storageAccount.isEmpty) Some(keyVaultParameters.account) else storageAccount
+            val secret = if (storageSecret.isEmpty) Some(keyVaultParameters.secret) else storageSecret
+            val container = if (storageContainer.isEmpty) Some(keyVaultParameters.container) else storageContainer
 
-          getAndValidateTransientStorageParameters(account, container, secret, storageSecretIsAccountKey = true)
+            tryGetAndValidateTransientStorageParameters(account, container, secret, storageSecretIsAccountKey = true)
+          }
         }
       } else {
-        KustoStorageParameters(storageAccount.get, storageSecret.get, storageContainer.get, storageSecretIsAccountKey)
+        Some(KustoStorageParameters(storageAccount.get, storageSecret.get, storageContainer.get, storageSecretIsAccountKey))
       }
     }
   }
 
-  private[kusto] def getAndValidateTransientStorageParameters(storageAccount: Option[String],
-                                                              storageContainer: Option[String],
-                                                              storageAccountSecret: Option[String],
-                                                              storageSecretIsAccountKey: Boolean): KustoStorageParameters = {
+  private[kusto] def tryGetAndValidateTransientStorageParameters(storageAccount: Option[String],
```

Comment:
i would revert the renaming
usually in kusto a method with try prefix i surrounding the implementation with try catch- but here we throw

Reply:
Changed to: 
getAndValidateTransientStorageParametersIfExist